sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -358,3 +359,51 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def check_indicator(X, expected_imputed_features, axis):
+    n_samples, n_features = X.shape
+    imputer = Imputer(missing_values=-1, strategy='mean', axis=axis)
+    imputer_with_in = clone(imputer).set_params(add_missing_indicator=True)
+    Xt = imputer.fit_transform(X)
+    Xt_with_in = imputer_with_in.fit_transform(X)
+    imputed_features_mask = X[:, imputer_with_in.imputed_features_] == -1
+    n_features_new = Xt.shape[1]
+    n_imputed_features = len(imputer_with_in.imputed_features_)
+    assert_array_equal(imputer_with_in.imputed_features_,
+                       expected_imputed_features)
+    assert_equal(Xt_with_in.shape,
+                 (n_samples, n_features_new + n_imputed_features))
+    assert_array_equal(Xt_with_in, np.hstack((Xt, imputed_features_mask)))
+    imputer_with_in = clone(imputer).set_params(add_missing_indicator=True)
+    assert_array_equal(Xt_with_in,
+                       imputer_with_in.fit_transform(sparse.csc_matrix(X))
```

Comment:
the shorthand `.A` is equivalent to `.toarray()` and is used often enough that you can put it here to avoid splitting over two lines.